framework/src/org/checkerframework/framework/flow/CFAbstractTransfer.java

```diff
@@ -1042,7 +1042,19 @@ protected void processConditionalPostconditions(
     @Override
     public TransferResult<V, S> visitCase(CaseNode n, TransferInput<V, S> in) {
         S store = in.getRegularStore();
-        return new RegularTransferResult<>(finishValue(null, store), store);
+        TransferResult<V, S> result =
+                new ConditionalTransferResult<>(
+                        finishValue(null, store), in.getThenStore(), in.getElseStore(), false);
+
+        V caseValue = null;
+        if (n.getCaseOperand()
+                != null) { // When encountering a default node, getCaseOperand returns null.
```

Comment:
I debugged through this and n.getCaseOperand() is never null.  Looking at the code that creates the CaseNodes (org.checkerframework.dataflow.cfg.CFGBuilder.CFGTranslationPhaseOne.SwitchBuilder#buildCase),  a CaseNode isn't created for the default case.